src/SSHLibrary/abstractclient.py

```diff
@@ -1002,17 +1012,17 @@ def _isabs(path):
         return destination.rsplit(path_separator, 1)
 
     def _create_missing_remote_path(self, path):
-        if path.startswith('/'):
-            current_dir = '/'
+        if path.startswith(b'/'):
+            current_dir = b'/'
         else:
-            current_dir = self._absolute_path('.')
-        for dir_name in path.split('/'):
+            current_dir = self._absolute_path(b'.').encode(self._encoding)
```

Comment:
yes, it return unicode string before:
```
    def _absolute_path(self, path):
        path = path.encode(self._encoding)
        if not self._is_windows_path(path):
            path = self._client.normalize(path)
        if not isinstance(path, unicode):
            path = unicode(path, self._encoding)
        return path
```
for javaclient:
```
    def _absolute_path(self, path):
        return self._client.canonicalPath(path)
```

Reply:
Hmm, `_absolute_path` requiring Unicode but returning bytes is definitely strange. It probably would be safest to change it to accept either Unicode or bytes but always return explicitly either Unicode or bytes. I would also assume it would be best for it to always return bytes, but that depends on its usage. At least in this case the return value is converted to bytes so returning bytes would make more sense.

The main reason this is messed up is that in Python 2 it didn't matter were Unicode or bytes accepted/returned as long as non-ASCII characters weren't involved. Making sure we accept/return correct types is probably easiest with unit tests. Haven't looked at SSHLibrary unit tests for ages, but it would be great if `_absolute_path` could be unit tested somewhat easily.